src/server/webkit.ts

```diff
@@ -88,143 +87,112 @@ export class WebKit extends BrowserTypeBase {
+        }
+        data.socket.send(JSON.stringify(message));
         return;
       }
-      socket.send(JSON.stringify(message));
-      return;
-    }
-    if (method === 'Playwright.pageProxyCreated') {
-      const socket = browserContextIds.get(params.pageProxyInfo.browserContextId);
-      if (!socket || socket.readyState === ws.CLOSING) {
-        // Drop unattributed messages on the floor.
+      if (method === 'Playwright.pageProxyCreated') {
+        const socket = browserContextIds.get(params.pageProxyInfo.browserContextId);
+        if (!socket || socket.readyState === ws.CLOSING) {
+          // Drop unattributed messages on the floor.
+          return;
+        }
+        pageProxyIds.set(params.pageProxyInfo.pageProxyId, { socket });
+        socket.send(JSON.stringify(message));
         return;
       }
-      pageProxyIds.set(params.pageProxyInfo.pageProxyId, socket);
-      socket.send(JSON.stringify(message));
-      return;
-    }
-    if (method === 'Playwright.pageProxyDestroyed') {
-      const socket = pageProxyIds.get(params.pageProxyId);
-      pageProxyIds.delete(params.pageProxyId);
-      if (socket && socket.readyState !== ws.CLOSING)
-        socket.send(JSON.stringify(message));
-      return;
-    }
-    if (method === 'Playwright.provisionalLoadFailed') {
-      const socket = pageProxyIds.get(params.pageProxyId);
-      if (socket && socket.readyState !== ws.CLOSING)
-        socket.send(JSON.stringify(message));
-      return;
-    }
-  };
+      if (method === 'Playwright.pageProxyDestroyed') {
+        const data = removePageProxy(params.pageProxyId);
+        if (data && data.socket.readyState !== ws.CLOSING)
+          data.socket.send(JSON.stringify(message));
+        return;
+      }
+      if (method === 'Playwright.provisionalLoadFailed') {
+        const data = pageProxyIds.get(params.pageProxyId);
+        if (data && data.socket.readyState !== ws.CLOSING)
+          data.socket.send(JSON.stringify(message));
+        return;
+      }
+    },
 
-  transport.onclose = () => {
-    for (const socket of sockets) {
-      socket.removeListener('close', (socket as any).__closeListener);
-      socket.close(undefined, 'Browser disconnected');
-    }
-    server.close();
-    transport.onmessage = undefined;
-    transport.onclose = undefined;
-  };
-
-  server.on('connection', (socket: ws, req) => {
-    if (req.url !== '/' + guid) {
-      socket.close();
-      return;
-    }
-    sockets.add(socket);
+    onClientAttached(socket: ws) {
+    },
 
-    socket.on('message', (message: string) => {
-      const parsedMessage = JSON.parse(Buffer.from(message).toString());
-      const { id, method, params } = parsedMessage;
-      const seqNum = idMixer.generate({ id, socket });
-      transport.send({ ...parsedMessage, id: seqNum });
+    onClientRequest(socket: ws, message: ProtocolRequest) {
+      const { method, params } = message;
+      const id = server.sendMessageToBrowser(message, socket);
```

Comment:
And here 'id' means something else. I guess I don't understand the convention...

Reply:
See, you also got confused because I left one seqNum named id...